api/books.ts

```diff
@@ -0,0 +1,188 @@
+  return await baseWithAuth()
+    .request({
+      url: "/api/user_books",
+      method: "POST",
+      data: {
+        user_book: {
+          book_id: id,
+          start_date: new Date().toISOString(),
+        },
+      },
+    })
+    .then((res) => {
+      const book = res.data
+      return book
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const getBookList = async (): Promise<BookType[]> => {
+  return await baseWithAuth()
+    .request({
+      url: "/api/books",
+      method: "GET",
+    })
+    .then((res) => {
+      const books = res.data
+      return books
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const getDiscover = async (): Promise<BookType[]> => {
+  return await baseWithAuth()
+    .request({
+      url: "/api/discover",
+      method: "GET",
+    })
+    .then((res) => {
+      const books = res.data
+      return books
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const getFinishedList = async (): Promise<BookType[]> => {
+  return await baseWithAuth()
+    .request({
+      url: "/api/user_books",
+      method: "GET",
+      params: {
+        finish_date: 1,
+      },
+    })
+    .then((res) => {
+      const books = res.data
+      return books
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const markAsRead = async (id: number): Promise<BookType> => {
+  return await baseWithAuth()
+    .request({
+      url: `/api/user_books/${id}`,
+      method: "PATCH",
+      data: {
+        user_book: {
+          finish_date: new Date().toISOString(),
+        },
+      },
+    })
+    .then((res) => {
+      const book = res.data
+      return book
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const getReadingList = async (): Promise<BookType[]> => {
+  return await baseWithAuth()
+    .request({
+      url: "/api/user_books",
+      method: "GET",
+    })
+    .then((res) => {
+      const books = res.data
+      return books
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const removeFromList = async (id: number): Promise<BookType> => {
+  return await baseWithAuth()
+    .request({
+      url: `/api/user_books/${id}`,
+      method: "DELETE",
+    })
+    .then((res) => {
+      const book = res.data
+      return book
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const setNotes = async (
+  id: number,
+  notes: string
+): Promise<BookType> => {
+  return await baseWithAuth()
+    .request({
+      url: `/api/user_books/${id}`,
+      method: "PATCH",
+      data: {
+        user_book: {
+          notes: notes,
+        },
+      },
+    })
+    .then((res) => {
+      const book = res.data
+      return book
+    })
+    .catch((error) => {
+      errorHandler(error)
+    })
+}
+
+export const setRating = async (
+  id: number,
+  rating: number | null
```

Comment:
Is there a scenario wherein ur function accepts a null rating?

Reply:
woops, I was wondering why and I forgot about this one lang pala when I was trying stuff out 😅

removed it na